benchmark-script/write_operations/main.go

```diff
@@ -29,47 +28,59 @@ var (
 	fFileSize = flag.Int("file-size", 1, "in KB")
 
 	fNumOfWrite = flag.Int("write-count", 1, "number of write iteration")
+	fOutputDir  = flag.String("output-dir", "", "Directory to dump the output")
 )
 
-func openFile(index int) (err error) {
-	fileName := path.Join(*fDir, "file_"+strconv.Itoa(index))
-	fileHandle, err := os.OpenFile(fileName, os.O_WRONLY|os.O_CREATE|os.O_TRUNC|syscall.O_DIRECT, 0644)
+var gResult *Result
+
+func init() {
+	gResult = &Result{}
+}
+
+func openFile(fileSuffix string) (fileHandle *os.File, err error) {
+	fileName := path.Join(*fDir, "file_"+fileSuffix)
+	fileHandle, err = os.OpenFile(fileName, os.O_CREATE|os.O_RDWR|os.O_TRUNC|syscall.O_DIRECT, 0644)
 	if err != nil {
 		err = fmt.Errorf("while opening file: %w", err)
 		return
 	}
-	fileHandles[index] = fileHandle
+
 	return
 }
 
 // Expect file is already opened, otherwise throws error
 func overWriteAlreadyOpenedFile(index int) (err error) {
 	for cnt := 0; cnt < *fNumOfWrite; cnt++ {
+		writeStart := time.Now()
+		fileName := strconv.Itoa(index) + "__" + strconv.Itoa(cnt)
+		fh, err := openFile(fileName)
+		if err != nil {
+			return fmt.Errorf("Error while creating the file %v", err)
+		}
+
 		for i := 0; i < (*fFileSize / *fBlockSize); i++ {
 			b := make([]byte, *fBlockSize*OneKB)
 
-			startByte := int64(i * (*fBlockSize * OneKB))
-
 			_, err = rand.Read(b)
```

Comment:
Should we move the byte generation code on the top? - As generation of random bytes is a cpu intensive operation.